components/ChallengeMaterial.tsx

```diff
@@ -139,35 +147,137 @@ export const ChallengeTitleCard: React.FC<ChallengeTitleCardProps> = ({
   )
 }
 
-type ChallengeQuestionCardProps = {
+const ChallengeQuestionCardDisplay: React.FC<{
   currentChallenge: ChallengeSubmissionData
-}
-export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
-  currentChallenge
-}) => {
-  const diff = _.get(currentChallenge, 'submission.diff', '')
-  const comment = _.get(currentChallenge, 'submission.comment', '')
-  const updatedAt = _.get(currentChallenge, 'submission.updatedAt', Date.now())
+}> = ({ currentChallenge }) => {
+  const comment = currentChallenge.submission?.comment
+  const updatedAt = currentChallenge.submission?.updatedAt || ''
   const context = useContext(GlobalContext)
   const name = context.session?.user?.name
   const username = context.session?.user?.username
-  const reviewerUserName = _.get(
-    currentChallenge,
-    'submission.reviewer.username',
-    null
-  )
+  const userId = context.session?.user?.id
+
+  const [index, setIndex] = useState(-1)
+
+  const [submission, setSubmission] = useState(currentChallenge.submission)
+  const reviewerUserName = submission?.reviewer?.username || null
   const comments =
-    currentChallenge.submission &&
-    currentChallenge.submission.comments?.filter(comment => !comment.line)
+    submission && submission.comments?.filter(comment => !comment.line)
```

Comment:
Could you remove the `submission && ....`  and just do `submission?.comments?.filter(comment => !comment.line)`

Reply:
Since we are using lodash, I think using this is better to `_.get` lodash function to maintain codebase consistency. With lodash, I'm not sure this repo needs to use the `?.` operator.
```
_.get(submission, 'comments').filter(...)
```

Reply:
Tom is right, looking at this part I am not sure why I needed `submissions&&` check in the first place.

Chaining `?` feels more natural than using lodash in this case. If anything we should start replacing `_.get` with `?` operators everywhere.

Reply:
Agreed on using the coalescing chain. It is more natural and works better with typescript. Lodash uses a string for the property which may cause bugs if the name changes for some reason.

Reply:
Ya I would vote moving to the coalescing operator when possible. A quick global search on the project for `_.get(` returns 40 hits in 13 files, seems like we should be able to refactor it out of the codebase at some point in the future. <br> \* This could be a very jr. opinion, as I haven't seen how bad projects can get when the consistency slowly gets destroyed with promises of future refactoring.

Reply:
Agreed on sprint Honeydew to use and replace `_.get` with the coalescing chain operator `?.`.